packages/test/test-utils/src/TestSummaryUtils.ts

```diff
@@ -131,6 +131,35 @@ export async function createSummarizer(
 	return createSummarizerCore(container, loader, summaryVersion);
 }
 
+/**
+ * Creates a summarizer client from the given container and returns the summarizer client's IContainer and ISummarizer.
+ * The ISummarizer can be used to generate on-demand summaries. The IContainer can be used to fetch data stores, etc.
+ *
+ * Can pass in a test config provider to enable/disable features.
+ */
+export async function createSummarizerWithTestConfig(
```

Comment:
I would not add a new function here. It's pretty much the same as `createSummarizer` except it adds `summaryOptions`. Can you modify `createSummarizer` to pass in summaryOptions instead?

Reply:
Ideally, you should replace `gcOptions` in `createSummarizer` with `IContainerRuntimeOptions` but that means you have to modify a whole bunch of tests. So, maybe add `summaryOptions` as the last param and then create an issue to update it and / or do it as a follow up.

Reply:
I'm not that opinionated here (as in I don't mind just tacking on a parameter to the existing one), but I think the new function is strictly better than the existing one. The existing function looks like it was made with GC in mind, but makes a pattern of adding a parameter for tacking on any feature flags that need to be turned on and isn't useful for other tests that might leverage it. For instance, if I wanted to use this function I would need to add `enableRuntimeIdCompressor` as a parameter, add it to the configuration object created in the method, and then calling the function would require me passing in undefined four times: 

`createSummarizer(provider, container, undefined, undefined, undefined, undefined, true)`

The difference here is that you can pass in an `ITestContainerConfig`, which most of our tests use and allows the tests to use their configuration without having to do this.

Reply:
Totally agree that this was written with GC in mind and someone moved it to test utils :-).
I also agree that this is better than the existing one. So, I would replace the existing one with this. I don't see value in having both.
You don't need to remove the existing one in this PR. Just create an issue when you PR merges so we are tracking it.